randovania/generator/dock_weakness_distributor.py

```diff
@@ -163,6 +163,7 @@ def __init__(self, game: GameDescription, configuration: BaseConfiguration, dock
 
     @classmethod
     def from_logic(cls, logic: Logic, dock: DockNode, target: DockNode) -> Self:
+        assert logic.game is not None
```

Comment:
if this were None, then wouldn't the call to `from_logic` fail since the type is specified as Logic?